File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/SubscriptionRecommendationResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import com.ssafy.e106.domain.promotion.dto.response.PromotionRecommendationSummary;
import com.ssafy.e106.domain.promotion.dto.response.PromotionRecommendationServiceResponse;
import com.ssafy.e106.domain.promotion.enums.PromotionType;
import java.util.List;

public record SubscriptionRecommendationResponse(
    Long promotionId,
    PromotionType promotionType,
    String title,
    String headline,
    Integer monthlySavingAmount,
    String billingCycle,
    String imageUrl,
    List<PromotionRecommendationServiceResponse> services) {

  public static SubscriptionRecommendationResponse from(PromotionRecommendationSummary summary) {
    return new SubscriptionRecommendationResponse(
        summary.promotionId(),
        summary.promotionType(),
        summary.title(),
        summary.headline(),
        summary.monthlySavingAmount(),
        summary.billingCycle(),
        summary.imageUrl(),
        summary.services());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionRecommendationResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionRecommendationResponse(
    val promotionId: Long,
    val promotionType: String,
    val title: String,
    val monthlySavingAmount: Int? = null,
    val billingCycle: String? = null,
    val headline: String,
    val imageUrl: String? = null,
    val services: List<SubscriptionRecommendationServiceResponse> = emptyList(),
)

@Serializable
data class SubscriptionRecommendationServiceResponse(
    val serviceId: Long,
    val serviceName: String,
    val logoUrl: String? = null,
)
